pkg/selectors/binaryentry.go

```diff
@@ -27,7 +35,21 @@ func (v *BinaryMapValue) NewValue() bpf.MapValue         { return &BinaryMapValu
 func (v *BinaryMapValue) GetValuePtr() unsafe.Pointer    { return unsafe.Pointer(v) }
 func (v *BinaryMapValue) DeepCopyMapValue() bpf.MapValue { return &BinaryMapValue{} }
 
-func writeBinaryMap(id int, path string, m *bpf.Map) error {
+func WriteBinaryMap(id uint32, path string) error {
+	mapDir := bpf.MapPrefixPath()
+	m, err := bpf.OpenMap(filepath.Join(mapDir, base.NamesMap.Name))
+	for i := 0; err != nil; i++ {
+		m, err = bpf.OpenMap(filepath.Join(mapDir, base.NamesMap.Name))
+		if err != nil {
+			time.Sleep(mapRetryDelay * time.Second)
+		}
+		if i > maxMapRetries {
+			panic(err)
```

Comment:
Let's not panic and crush the agent. We can return an error instead.

Reply:
for what its worth seems we have a couple other panics from similar errors.